keylime/ca_impl_openssl.py

```diff
@@ -188,5 +193,9 @@ def gencrl(serials, cert: str, ca_pk: str):
         cert_builder = cert_builder.revocation_date(date_now)
         builder = builder.add_revoked_certificate(cert_builder.build())
 
+    if not isinstance(
+        priv_key, (EllipticCurvePrivateKey, RSAPrivateKey, DSAPrivateKey, Ed448PrivateKey, Ed25519PrivateKey)
+    ):
+        raise ValueError(f"Unsupported key type {type(priv_key).__name__}")
```

Comment:
Have you checked on how the callers are handling this ValueError?

Reply:
If I read the code correctly, no one is capturing this, so the user will see the exception and can fix the issue